### ISSUE TITLE
Gin has no error types

### ISSUE THREAD

**Issue by Guillembonet** [oapi-codegen/oapi-codegen#1682]
I see that other servers like chi or std return different error types such as `RequiredHeaderError` or `InvalidParamFormatError`, but gin returns errors without a specific type like `fmt.Errorf("Query argument page is required, but not found")`. This makes errors handling code more uglier and more prone to bugs.

Is this for some reason desired behavior?

**Comment by jamietanna** [comment]
Nope, this looks like a gap in the templates - Gin was an earlier implementation than Chi and Gorilla - from memory - so could be we introduced it as a pattern after the fact. Contributions welcome!

**Comment by Guillembonet** [comment]
Okay, I might try to do a PR for this next week! Thanks!
